test/coordinate_unit_test.js

```diff
@@ -0,0 +1,56 @@
+/* global describe, it */
+import proj4 from 'proj4';
+import assert from 'assert';
+import Coordinates, { UNIT } from '../src/Core/Geographic/Coordinates';
+
+// Define projection that we will use (taken from https://epsg.io/3946, Proj4js section)
+proj4.defs('EPSG:3946', '+proj=lcc +lat_1=45.25 +lat_2=46.75 +lat_0=46 +lon_0=3 +x_0=1700000 +y_0=5200000 +ellps=GRS80 +towgs84=0,0,0,0,0,0,0 +units=m +no_defs');
+
+// Asster two float number are equals, with 5 digits precision.
+function assertFloatEqual(float1, float2) {
+    assert.equal(Number(float1).toFixed(5), Number(float2).toFixed(5));
+}
+
+// Assert two coordinates obects are equals.
+function assertCoordEqual(coord1, coord2) {
+    assert.equal(coord1._internalStorageUnit, coord2._internalStorageUnit);
+    assertFloatEqual(coord1._values[0], coord2._values[0]);
+    assertFloatEqual(coord1._values[1], coord2._values[1]);
+    assertFloatEqual(coord1._values[2], coord2._values[2]);
+}
+
+describe('Coordinate conversions', function () {
+    this.timeout(10000);
+    it('Coordinate conversion from EPSG:4326 to EPSG:4978 and back to EPSG:4326', function (done) {
+        var coord1 = new Coordinates('EPSG:4326', 15, 12);
+        var coord2 = coord1.as('EPSG:4978');
+        var coord3 = coord2.as('EPSG:4326');
+        assertCoordEqual(coord1, coord3);
+        done();
+    });
+    it('Coordinate conversion from EPSG:3946 (Lyon WFS bus example) to EPSG:4978 (Globe) and back to EPSG:3946', function (done) {
+        var coord1 = new Coordinates('EPSG:3946', 1500000, 5100000, 12);
+        var coord2 = coord1.as('EPSG:4978');
```

Comment:
can we assert coord2 too? You can use epsg.io to get the expected value in EPSG:4978.

Reply:
Good idea, but 
- epsg.io doesn't work with an altitude input (here it's 12)
- and epsg.io doesn't give Z value in output, so the coordinate comparison fail.

Reply:
And I's not the purpose of this test : In this test, I want to convert a coordinate to another crs, and convert back to the first crs. I prefer this test to be simple and easy to read like this.

Reply:
> epsg.io doesn't work with an altitude input (here it's 12)

I'm sure there's other way of finding converted altitude :-)

> And I's not the purpose of this test : In this test, I want to convert a coordinate to another crs, and convert back to the first crs. I prefer this test to be simple and easy to read like this.

The test would also pass if `.as` does absolutely nothing, that's why I thing it's more robuts if we also test that it *actually* do something. 

For a start, you could just assert on x and y.

Reply:
Note that this remark is also true for the other tests, some of which do not set a z.

Reply:
as I want to keep the tests simple,
I added another test to test simple conversion to another crs and verify values.